src/libraries/Microsoft.VisualBasic.Core/src/Microsoft/VisualBasic/CompilerServices/NewLateBinding.vb

```diff
@@ -927,6 +942,18 @@ Namespace Microsoft.VisualBasic.CompilerServices
                 baseReference = New Container(Instance)
             End If
 
+            If baseReference.IsCOMObject And Not baseReference.IsWindowsRuntimeObject Then
+                Try
+                    LateBinding.InternalLateSet(Instance, Type, MemberName, Arguments, ArgumentNames, OptimisticSet, CallType)
+                    If (RValueBase And Type.IsValueType) Then
```

Comment:
>Type.IsValueType [](http://example.com/codeflow?start=39&length=16)

Why use `Type.IsValueType` here rather than `baseReference.IsValueType` as used elsewhere?

Reply:
The `Type` is a byref to `InternalLateSet` and can be updated.

Reply:
Do I need to mark something to properly pass a value byref to a method in VisualBasic?

Reply:
> The `Type` is a byref to `InternalLateSet` and can be updated.

Thanks. Consider adding a comment to indicate that, perhaps:
`' Using Type.IsValueType rather than baseReference.IsValueType since Type may be modified in InternalLateSet.`